library/command/qsl_command/src/main/java/org/quiltmc/qsl/command/api/CommandRegistrationCallback.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.command.api;
+
+import com.mojang.brigadier.CommandDispatcher;
+import net.minecraft.server.command.ServerCommandSource;
+import org.quiltmc.qsl.base.api.event.ArrayEvent;
+
+/**
+ * Event for registering server-side commands.
+ */
+@FunctionalInterface
+public interface CommandRegistrationCallback {
+	/**
+	 * Invoked when server-side commands are registered.
+	 */
+	ArrayEvent<CommandRegistrationCallback> EVENT = ArrayEvent.create(CommandRegistrationCallback.class, callbacks -> (dispatcher, integrated, dedicated) -> {
```

Comment:
If a server is not dedicated it is integrated - no need for an extra parameter

Reply:
Not always the case - there are debug and datagen situations where both the integrated and dedicated parameters are true.